core/src/control/dihu_context.cpp

```diff
@@ -495,6 +512,14 @@ int DihuContext::ownRankNo() { return rankSubset_->ownRankNo(); }
 
 int DihuContext::nRanksCommWorld() { return nRanksCommWorld_; }
 
+std::shared_ptr<Checkpointing::Handle> DihuContext::getCheckpointing() const {
```

Comment:
Returning shared_ptr can have pitfalls. I think it would be safer to pass 'Checkpointing::Handle' object by-reference and then perform assignment operation.